### ISSUE TITLE
Files pattern should include confirmation step when deleting

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/vets-design-system-documentation#1770]
### Duplicate check

- [X] I've searched for any related issues and avoided creating a duplicate issue.

### This update is for:

Pattern

### What is the name?

Files

### What is the nature of this update?

- [X] How to build this component/pattern
- [ ] When to use this component/pattern
- [ ] When to use something else
- [ ] Usability guidance
- [X] Accessibility
- [X] Implementation
- [ ] Research insights
- [ ] Package information
- [ ] Addition to Forms Library documentation
- [X] Update to existing Forms Library documentation

### What problem does this solve?

The [Files pattern](https://design.va.gov/patterns/ask-users-for/files) includes examples for:
- Default state (no file uploaded)
- Loading state (upload in progress)
- Review state (file(s) have been uploaded, summary information displayed)

I would like to add a "Delete state" that shows the preferred pattern for deleting an uploaded file. In this pattern, users would be prompted to confirm the delete action (with an option to cancel) before the file is actually deleted. I think the most straightforward implementation of this would be with a modal, but I'm open to other options as well!

Additionally, I would like to add the following to the accessibility guidance on the pattern (please edit as appropriate!):
> - **Ask for confirmation when deleting files.** Destructive actions like deleting files should require two steps by users --- the initial button click, and then a confirmation. This helps prevent users from accidentally deleting a file with an unintentional click, and provides an extra prompt for screen reader users and screen magnification users who might not see the visual change when a file is removed.

### Additional Context

This came up in a [staging review for Supplemental Claims Form 20-0995](https://github.com/department-of-veterans-affairs/va.gov-team/issues/56082), and @Mottie is [going to be working on it](https://github.com/department-of-veterans-affairs/va.gov-team/issues/58596) and may be able to provide some good screenshots. 😄 

See also [WCAG SC 3.3.4](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html) and [WCAG Technique G168](https://www.w3.org/WAI/WCAG21/Techniques/general/G168). A form that doesn't confirm file deletion probably isn't technically a 3.3.4 violation since users will have an opportunity to review before final submission... but it definitely falls under the spirit of 3.3.4 and would be good to address.

**Comment by Mottie** [comment]
I've updated the `FileField` component, but I didn't switch the input to a `va-file-input` because React Testing Library (RTL) doesn't work well with shadow DOM - The PR was already huge enough updating the React component to a functional component, and updating the unit tests to RTL. Maybe I'll tackle the change to `va-file-input` in the future.

Also note: File upload code outside of platform (COE & Claim Status Tool) have not yet been updated.

**Comment by Mottie** [comment]
@briandeconinck Josh is on PTO until 7/24, if you would please test in his place? Our forms that have uploads are:
* [Supplemental Claim](https://staging.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995/start)
* [Request a Board Appeal](https://staging.va.gov/decision-reviews/board-appeal/request-board-appeal-form-10182/introduction) (NOD)

You can log in to user 228 (Colder). Note that non of our mock users currently include any (loaded) contestable issues in staging.

**Comment by Mottie** [comment]
> Would love for them to be "Delete [filename]" or have an aria-label to make it easier for screen reader users to differentiate the files.

Hmmm, the outer `va-button` includes a `aria-describedby` pointing to the filename

```html
<va-button
  secondary
  class="delete-upload vads-u-width--auto hydrated"
  aria-describedby="root_additionalDocuments_file_name_0"
  text="Delete file" />
```

I guess it doesn't work on the outer wrapper. I'll make a quick update to add a `label` (adds an `aria-label` to the internal button) instead. Thanks for looking!

**Comment by Mottie** [comment]
@briandeconinck Please test the button label updates 😸

**Comment by Mottie** [comment]
Nudge @briandeconinck - this work _should_ be done. Please test at your convenience

**Comment by humancompanion-usds** [comment]
@allison0034 - May I trouble you to gather a screenshot or 2 for this issue? Thanks! I'm making the guidance update now.

**Comment by Mottie** [comment]
Will the [screenshots from this PR work](https://github.com/department-of-veterans-affairs/vets-website/pull/24787)?

**Comment by allison0034** [comment]
@Mottie I think that would work but the per the [content considerations](https://design.va.gov/patterns/ask-users-for/files#content-considerations) for that component we should use file instead of document. And after looking at the other screen shots in our DS most of them have documents. @humancompanion-usds  I don't want to scope creep, but can we replace those as well?

**Comment by allison0034** [comment]
@humancompanion-usds would these work? They are from a Sketch file. This team used a white card instead of gray also. 
<img width="342" alt="Screenshot 2024-01-09 at 10 39 17 AM" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/72398073/a5b5092a-67a0-47b9-8432-8cc1019b7d57">
<img width="361" alt="Screenshot 2024-01-09 at 10 39 37 AM" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/72398073/2cd4126b-7bad-48a2-9546-76128a7234ed">

**Comment by humancompanion-usds** [comment]
@allison0034 - If this has shipped and we're sure of that then yes I don't mind using the Sketch file. However, we want to also show the modal that opens when you click "Delete file". Is that in there as well?

**Comment by Mottie** [comment]
Oh, I think the content was changed from "file" to "document" - the screenshots are likely out of date

**Comment by allison0034** [comment]
@humancompanion-usds I worked with Robin to get some screens that have the correct content and the modal in production. Let me know if you need more.
![1-file-upload_ReqBoardApp](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/72398073/5d2963e6-4559-42f3-89ba-f244a71a1628)
![1-file-upload_SuppClaim](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/72398073/bd725932-bbbe-4038-8bc8-19fcd361e3cc)
<img width="509" alt="2-file-upload" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/72398073/7e4b048c-237d-452d-b764-e3e4f651e5e0">
![2-file-upload-SuppClaim](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/72398073/b0809fbf-d8a0-4103-8538-32c6d2ea738d)
<img width="640" alt="3-2-file-upload-SuppClaim" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/72398073/ddb6b117-2f66-4e22-a502-7a972ad01b96">
![3-file-upload-SuppClaim](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/72398073/a8071dc3-8e3c-4778-acaa-f59a1f123c29)
